Code/GraphMol/SmilesParse/SmilesParse.cpp

```diff
@@ -374,6 +374,7 @@ RWMol *SmilesToMol(const std::string &smiles,
   if (res && params.allowCXSMILES && !cxPart.empty()) {
     std::string::const_iterator pos = cxPart.cbegin();
     SmilesParseOps::parseCXExtensions(*res, cxPart, pos);
+    res->setProp("_CXSMILES_Data", cxPart.substr(0, pos - cxPart.cbegin()));
```

Comment:
We generally try and avoid "magic strings" to help prevent hard to find bugs.  Could you add a CXSMILES_Data entry to

Code/RDGeneral/types.h/cpp

and use that instead.  This isn't exposed to python (yet) by the way, I should really make an issue for that.

Reply:
Side note, string.substr is a bit redundant, you can avoid the use of substr with

```
setProp(CXSMILES_Data ,std::string(pos, cxPart.end()));
```

I have no preference, by the way, this is just an observation.

Reply:
Though I agree that it's useful to have these string constants defined in a central place, I think we should consider moving them out of `types.h` and making them more local. `types.h` is included in a bunch of source files, so adding a new property name to it effectively results in doing a `make clean`. And that's no fun at all.

I will put a bit of thought into this and come up with a proposal.

Reply:
In the meantime, I don't think it's tragic to have the name explicit in the code, so I'd be ok with merging this as is.

Reply:
On the `substr` comment. I don't think it's at all redundant here.
Your suggestion copies from `pos` to the end of the string.
The code here copies from the beginning of the string to `pos`.

Reply:
Yes, I should have written

string(cxPart.begin(), pos)

The point I was making is that we already have the iterators so substr is doing a bit more work and is kind of a redundant api anyway.

As for types.h I have no real issue separating these into there dependent modules as long as we then expose them ( as least the public ones ) to python.  I’ve been meaning to expose them for a while.

What do you feel about making a specialized struct in each library ala

struct GraphMolProps {
  std::string atommapno:
};
struct FileParserProps...

extern GraphMolProps graphprops;

Or perhaps
const GraphMolProps &GetGraphMolProps();

Now that we are using c++11

Reply:
seems like having them in structs like this makes sense...

Reply:
Since `pos` is a const_iterator, `cxPart.cbegin()`.

(Pointing out only because I was feeling like being left out of the discussion :D )